types/wlr_compositor.c

```diff
@@ -44,13 +52,13 @@ static void wl_compositor_create_region(struct wl_client *client,
 	wlr_region_create(client, resource, id);
 }
 
-struct wl_compositor_interface wl_compositor_impl = {
+static const struct wl_compositor_interface wl_compositor_impl = {
 	.create_surface = wl_compositor_create_surface,
 	.create_region = wl_compositor_create_region
 };
 
 static void wl_compositor_destroy(struct wl_resource *resource) {
-	struct wlr_compositor *compositor = wl_resource_get_user_data(resource);
+	struct wlr_compositor *compositor = compositor_from_resource(resource);
```

Comment:
Let's be consistent: wlr_compositor_from_resource.

Reply:
Static fonctions drop the wlr_ prefix.

Reply:
-1

Reply:
fwiw I see an argument for exporting these functions to the public API

Reply:
>-1

You mean static functions should have the wlr_ prefix? Not dropping the prefix for this particular function *is* inconsistent.

>fwiw I see an argument for exporting these functions to the public API

A few are exported (wlr_output, wlr_region, wlr_surface). We could export others, but not sure it makes sense (wlr_screenshooter, wlr_decoration_manager...).

Reply:
>You mean static functions should have the wlr_ prefix? Not dropping the prefix for this particular function is inconsistent.

I mean *these* static functions should have the wlr_ prefix. But maybe not. Let me think about it.